### PR TITLE
[HTTP] Catch and ignore exception from an external server

### LINKED ISSUE #87864: [TestFailure] GetAsync_SetAutomaticDecompression_ContentDecompressed_Deflate (Gateway Timeout)
## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=313503
Build error leg or test failing: System.Net.Http.Functional.Tests.HttpClientHandler_RemoteServerTest.GetAsync_SetAutomaticDecompression_ContentDecompressed_Deflate

## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-section).

```json
{
  "ErrorPattern": "System\\.Net\\.Http\\.Functional\\.Tests\\.HttpClientHandler_RemoteServerTest\\.GetAsync_SetAutomaticDecompression_ContentDecompressed_Deflate\\(uri: \"\"https?:\/\/httpbin\\.org\/deflate\"\", expectedContent: \"\"\\\\\"\"deflated\\\\\"\": true\"\"\\) \\[FAIL\\]\\n.*504",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

### PR DESCRIPTION
**Pull request by ManickaP**
Specific error codes pulled from kusto:
```kql
AzureDevOpsTests
| where TestName has "GetAsync_SetAutomaticDecompression_ContentDecompressed_Deflate"
| summarize max(RunCompleted) by Message
```
|Message|max_RunCompleted|
|-|-|
|System.Net.Http.HttpRequestException : Response status code does not indicate success: 504 (Gateway Time-out). |2023-07-25T13:06:00.503Z|
|System.Net.Http.HttpRequestException : Response status code does not indicate success: 502 (Bad Gateway).	|2023-07-25T12:09:21.28Z|
|System.Net.Http.HttpRequestException : Resource temporarily unavailable (httpbin.org:80)<br/>---- System.Net.Sockets.SocketException : Resource temporarily unavailable	|2023-07-04T12:47:25.46Z|
|System.Net.Http.HttpRequestException : Resource temporarily unavailable (httpbin.org:443)<br/>---- System.Net.Sockets.SocketException : Resource temporarily unavailable	|2023-07-01T12:42:15.28Z|

Fixes #87864

### CODE REVIEW

File: src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs [hunk 1]
```diff
@@ -1289,7 +1290,16 @@ public async Task GetAsync_SetAutomaticDecompression_ContentDecompressed_Deflate
             handler.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
             using (HttpClient client = CreateHttpClient(handler))
             {
-                Assert.Contains(expectedContent, await client.GetStringAsync(uri));
+                try
+                {
+                    Assert.Contains(expectedContent, await client.GetStringAsync(uri));
+                }
+                catch (HttpRequestException hre) when (hre.StatusCode == HttpStatusCode.GatewayTimeout || hre.StatusCode == HttpStatusCode.BadGateway ||
+                                                      (hre.InnerException is SocketException se && (se.SocketErrorCode == SocketError.WouldBlock || se.SocketErrorCode == SocketError.TryAgain)))
```
Comment (MihaZupan) [inline src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs:1298]:
```suggestion
                catch (HttpRequestException hre) when (hre.StatusCode is HttpStatusCode.GatewayTimeout or HttpStatusCode.BadGateway ||
                                                      (hre.InnerException is SocketException se && (se.SocketErrorCode is SocketError.WouldBlock or SocketError.TryAgain)))
```

**Review by MihaZupan — APPROVED** [review]

**Review by CarnaViire — APPROVED** [review]

File: src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs (hunk 1 above)
Comment (ManickaP) [reply src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs:1298]: Nice!

**Comment by ManickaP** [comment]
Failures unrelated.

**Pull request merged**